fud2/fud-core/src/run.rs

```diff
@@ -87,6 +87,77 @@ impl EmitBuild for EmitBuildFn {
     }
 }
 
+/// A config variable.
+#[derive(Debug, Clone)]
+pub enum ConfigVar {
+    /// The key for the config variable.
+    Var(String),
+    /// The key for the config variable followed by the value it should be if the key is not found.
+    VarOr(String, String),
+}
+
+/// The data required to emit a single op.
+pub struct OpEmitData {
+    /// The name of the op.
+    pub op_name: String,
+
+    /// The commands run whenever the op run. Each of these must be run in order in a context where
+    /// the variables in each cmd are supplied. In particular, this means that variables of the
+    /// form "$[i|o]<digit>" are in scope.
+    pub cmds: Vec<String>,
```

Comment:
Since we're doing quite a bit less of gluing stuff together with `&&`, maybe it would be simpler to just make this a `String`? If people really want to add multiple commands here, they can always concoct the `&&` themselves…

Reply:
I don't think so because I would like this struct to correspond to a single op. Ops can have multiple shell commands in their body so making this a string would imply moving the gluing back up a level of abstraction. That isn't necessarily bad, but I like a little more having that logic in the implementation of `EmitBuild`.